src/sentry/lang/javascript/processor.py

```diff
@@ -861,19 +861,36 @@ def is_valid_frame(frame):
     return frame is not None and frame.get("lineno") is not None
 
 
-def get_function_for_token(frame, token):
+def get_function_for_token(frame, token, previous_frame=None):
     """
-    Get function name for a given frame, based on the looked up token.
-    Return tokens name if we have a usable value from symbolic or we have no initial function name at all,
-    otherwise, fallback to frames current function name.
+    Get function name for a given frame based on the token resolved by symbolic.
+    It tries following paths in order:
+    - return token function name if we have a usable value (filtered through `USELESS_FN_NAMES` list),
+    - return caller (previous frame) token function name if it had one mapped,
```

Comment:
This should be:

> The SourceMapped `name` of the caller token.

```js
function caller(callee) {
  return callee();
  //           ^ stack frame points exactly here
}
caller(() => { throw new Error(); });
```

The stack trace points to the call expression `callee()` which should have a `name` mapping. That is what we want. We don’t want the function name (aka scope name) of the caller.